src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,454 @@
+                this.boxIsShow = false
+                this.undrawBox()
+            } else {
+                this.boxIsShow = true
+                this.drawBox()
+            }
+	}  else if (this.sketch.keyCode === this.sketch.RIGHT_ARROW) {
+        // Show/hide label when right arrow key is pressed
+    if (this.showLabel == false) {
+                this.showLabel = true
+                this.drawLabel()
+            } else {
+                this.showLabel = false
+                this.undrawLabel()
+            }
+	}
+	}
+	**/
+    keyboardEvents() {
+        // Show description box when the left arrow key is pressed
+        if (this.sketch.keyIsDown(this.sketch.LEFT_ARROW)) {
+            if (this.boxIsShow == true && this.subG != null) {
+                this.boxIsShow = false
+                this.undrawBox()
+            } else {
+                this.boxIsShow = true
+                this.drawBox()
+            }
+        }
+
+        // Show label when right arrow key is pressed
+        if (
+            this.sketch.keyIsDown(this.sketch.RIGHT_ARROW)
+            && this.firstDraw == false
+        ) {
+            if (this.showLabel == false) {
+                this.showLabel = true
+                this.drawLabel()
+            } else {
+                this.showLabel = false
+                this.undrawLabel()
+            }
+        }
+
+        // Increase brightness "[" key is pressed
+        if (this.sketch.keyIsDown(219) && this.firstDraw == false) {
+            this.brightnessUp()
+        }
+
+        // Decrease brightness "]" key is pressed
+        if (this.sketch.keyIsDown(221) && this.firstDraw == false) {
+            this.brightnessDown()
+        }
+    }
+
+    brightnessUp() {
+        this.brightAdjust += 1
+        this.redrawCircles()
+    }
+
+    brightnessDown() {
+        this.brightAdjust -= 1
+        this.redrawCircles()
+    }
+
+    // Draw labels for each circle
+    drawLabel() {
+        this.position = this.sketch.createVector(
+            this.initialPosition.x,
+            this.initialPosition.y
+        )
+        this.subL = this.sketch.createGraphics(
+            this.canvasSize.x,
+            this.canvasSize.y
+        )
+        this.subG.colorMode(this.sketch.HSB)
+        this.subG.noStroke()
+        for (let i = this.seq.first; i < this.n; i++) {
+            this.showCircleLabel(i)
+            this.changePosition()
+        }
+    }
+
+    // Remove all labels by drawing circles again
+    undrawLabel() {
+        this.redrawCircles()
+    }
+
+    redrawCircles() {
+        this.position = this.sketch.createVector(
+            this.initialPosition.x,
+            this.initialPosition.y
+        )
+        this.firstDraw = true
+        this.currentIndex = this.seq.first
+        this.sketch.redraw()
+    }
+
+    drawCircle(ind: number) {
+        const numberNowBigint = this.seq.getElement(ind)
+        const numberNow = Number(numberNowBigint)
```

Comment:
For discussion: should we try to make this visualizer work with bigint?  It allows for arbitrary formulas in terms of the input values based on @gwhitney 's code, so I don't know how feasible that is.

Reply:
we decided to document its failings; see [comment here](https://github.com/numberscope/frontscope/pull/231#issuecomment-1377967936)